### DISCUSSION TITLE
클린코드가 뭐라고 생각하는지?

### DISCUSSION THREAD

**Discussion by gyungchan-jo** [2024-woowacourse-study/level-interview#9] [Clean Code]
.

**Comment by gyungchan-jo** [comment]
클린코드란 가독성, 단순성, 유지보수성 등 특징을 갖춘 있는 코드를 의미한다고 생각합니다.

1. 가독성: 코드는 사람이 읽기 쉬워야 합니다. 변수, 함수, 클래스 등의 이름은 명확하고 의미 있게 지어야 합니다.
2. 단순성: 코드는 간결하고 단순해야 합니다. 복잡한 코드는 오류를 발생시키기 쉽고 유지 보수가 어렵습니다. 따라서 불필요한 복잡성을 최소화하고 간결한 설계를 지향해야 합니다.
3. 유지보수성: 코드는 변경에 유연하게 대응할 수 있어야 합니다. 새로운 기능 추가나 버그 수정과 같은 변경 사항에 대응하기 쉬워야 하며, 변경이 발생했을 때 다른 부분에 영향을 최소화해야 합니다.
4. 중복 제거: 코드 중복은 피해야 합니다. 중복된 코드는 수정이 번거로워지고 유지보수성이 저하됩니다. 따라서 중복을 최소화하고 코드를 재사용할 수 있는 구조를 유지해야 합니다.
5. 응집도와 결합도: 코드는 응집도를 높이고 결합도를 낮추는 방향으로 설계해야 합니다. 응집도가 높은 코드는 한 가지 기능에 집중되어 있어야 하며, 결합도가 낮은 코드는 다른 모듈과의 의존성을 최소화해야 합니다.

**Comment by le2sky** [comment]
기계가 이해할 수 있는 코드는 누구나 작성 가능합니다. 
하지만, 사람이 이해할 수 있는 코드는 잘 훈련된 소프트웨어 엔지니어만이 작성할 수 있습니다.
단순히 아름답고, 향기로운 설계를 하거나, 로버트 C. 마틴의 클린 코드에 존재하는 모든 내용을 엄격하게 적용했다고 해서, 클린 코드라고 보기 어렵다고 생각합니다.
결국 소프트웨어는 팀원과 함께 만들어가고, 그게 아니여도 미래(혹은 과거!?)의 자신과 만들어 갑니다.
읽는 사람을 생각하는 태도를 가지고, 오용하기 어려운 코드, 실수를 피하는 코드, 가독성이 높은 코드를 작성하는 노력이 필요합니다. 

(결론) Java라는 엔터프라이즈용 언어를 사용하고 있는 입장에서 클린코드는 결국 미래에 유지보수하기 쉽고, 이 코드를 읽을 개발자를 고려한 코드라고 생각합니다.

**Comment by GIVEN53** [comment]
협업, 코드를 읽는 것에 70% 시간 소비
-> 읽는 사람, 사용하는 사람 고려해야 함

가독성
오용하기 어려움, 실수 방지 -> 불변
